### ISSUE TITLE
Serial is hanging.

### ISSUE THREAD

**Issue by zap8600** [copy/v86#899]
Me and someone are working on a web demo for something. However, the system isn't receiving any serial data. We are working with a custom x86-16 bootloader. Here is the HTML code:
```
<!doctype html>
<title>v86 Emulator</title>

<script src="/libv86.js"></script>
<script>
"use strict";

window.onload = function()
{
    var data = "";

    var emulator = window.emulator = new V86Starter({
        wasm_path: "/v86.wasm",
        memory_size: 32 * 1024 * 1024,
        vga_memory_size: 2 * 1024 * 1024,
        screen_container: document.getElementById("screen_container"),
        bios: {
            url: "/bios/seabios.bin",
        },
        vga_bios: {
            url: "/bios/vgabios.bin",
        },
        fda: {
            url: "/sectorc.bin",
        },
        autostart: true,
    });

    emulator.add_listener("serial0-output-char", function(char)
    {
        if(char === "\r")
        {
            return;
        }

        data += char;
        document.getElementById("terminal").value += char;
    });

    var compile = function() {
        var fileInput = document.getElementById('source_code');
        var file = fileInput.files[0];
        var reader = new FileReader();

        reader.onload = function(event) {
          data = event.target.result;
          data += "\n";
          console.log(data);
        };

        reader.readAsText(file);
        emulator.serial0_send(data);
    }
    window.compile = compile;
}
</script>

<input type="file" id="source_code">

<button onclick="window.compile()">Compile</button>

<!-- A minimal structure for the ScreenAdapter defined in browser/screen.js -->
<div id="screen_container">
    <div style="white-space: pre; position: relative; font-family: Liberation Mono, DejaVu Sans Mono, Courier New, monospace; font-weight: bold; font-size: 15px; line-height: 14px"></div>
    <canvas style="display: none"></canvas>
</div>

<textarea readonly rows=25 cols=60 id="terminal"></textarea>
```
The code for the system is [here](https://github.com/xorvoid/sectorc/blob/main/sectorc.s).

**Comment by copy** [comment]
This fixes the input reading:

```js
        reader.onload = function(event) {
          data = event.target.result;
          data += "\n";
          console.log(data);
          reader.readAsText(file); // this line needs to be here
        };

        emulator.serial0_send(data);
```

According to v86's log, the input is now read correctly:

```
[...]
libv86-debug.js:7899 10:14:28+434 [SERI] input: 0xA
libv86-debug.js:7899 10:14:28+434 [SERI] input: 0x20
libv86-debug.js:7899 10:14:28+434 [SERI] input: 0x20
libv86-debug.js:7899 10:14:28+434 [SERI] input: 0x7D
libv86-debug.js:7899 10:14:28+434 [SERI] input: 0xA
libv86-debug.js:7899 10:14:28+434 [SERI] input: 0x7D
libv86-debug.js:7899 10:14:28+434 [SERI] input: 0xA
```

That said, I don't get any output in any of the examples, neither in qemu nor v86. I've reported the bug in the other repo: https://github.com/xorvoid/sectorc/issues/14

**Comment by copy** [comment]
I'll push a fix for this soon.

**Comment by copy** [comment]
@zap8600 Yeah, needed to include lib.c and _start.c, and a small fix on v86's side (0836a20d2f75118f0eac754f561883f42fc46d98).

I've uploaded a demo here: https://copy.sh/v86/examples/sectorc.html
